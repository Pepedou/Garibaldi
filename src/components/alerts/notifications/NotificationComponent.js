import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

class NotificationComponent extends Component {
  render() {
    let { notifications } = this.props;
    return notifications.length > 0 ? (
      <div className="Notifications">
        {notifications.map(({ type, contentType, messages, message }, key) => (
          <div className={`alert alert-${type}`} key={key}>
            {contentType === "list" ? (
              <div className="NotificationMessages">
                {messages.map((text, key) => (
                  <div className="NotificationMessage" key={key}>
                    {text}
                  </div>
                ))}
              </div>
            ) : (
              message
            )}
          </div>
        ))}
      </div>
    ) : null;
  }
}

NotificationComponent.displayName = "NotificationComponent";

export const mapStateToProps = ({ notifications }) => ({ notifications });

NotificationComponent.propTypes = {
  notifications: PropTypes.array
};

export default connect(mapStateToProps, null)(NotificationComponent);
