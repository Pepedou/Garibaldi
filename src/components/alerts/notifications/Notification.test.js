import NotificationComponent from "./NotificationComponent";
import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";

let store = {
  subscribe: () => {},
  dispatch: () => {},
  getState: () => {
    return { notifications: [] };
  }
};

it("NotificationComponent renders", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <NotificationComponent />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
