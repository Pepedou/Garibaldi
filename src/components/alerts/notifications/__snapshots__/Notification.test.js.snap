// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationComponent renders 1`] = `null`;
